astropy/units/decorators.py

```diff
@@ -252,7 +253,9 @@ def wrapper(*func_args, **func_kwargs):
                 return_ = wrapped_function(*func_args, **func_kwargs)
 
             valid_empty = (inspect.Signature.empty, None)
-            if wrapped_signature.return_annotation not in valid_empty:
+            if (wrapped_signature.return_annotation not in valid_empty) and isinstance(
+                wrapped_signature.return_annotation, (str, UnitBase, FunctionUnitBase)
+            ):
```

Comment:
Total nitpick, but could the closing parenthesis be on the previous line? I still find this aspect of `blacken` really unpythonic and cluttery...  (Also, beyond PEP8 our general request is to follow the style of existing code in a file)

Reply:
Instead of any string, the code should check for a valid physical type. `mypy` and similar allow for a string annotation to be used instead of a class, where the class has not yet been defined.

Reply:
Ideally, yes, but I think here I think it is fine to be simple. 

But, now thinking about it: since the return annotation should normally be a class if it is not a unit, could one just do `not isinstance(..., type)`?

Reply:
https://www.python.org/dev/peps/pep-0563/

Reply:
This won't be a problem starting in python 3.10, but it can be problematic here.

Reply:
Maybe, but then won't a PhyicalType instance annotation (#11204, @namurphy) trip the check?

Reply:
Sure, this was done via autoformatting and i didn't pay enough attention to keep the source code in line.

Reply:
Given the ongoing refactoring, I think we should keep the focus on quick & easy here! Would my `not isinstance(..., type)` be good enough? Or shall we just stick what's here now? Note that currently `PhysicalType` instances would fail too, so no need I think to take them into account specifically.

Reply:
It should be good enough 👍 

```
def func(x) -> int:
    pass

isinstance(func.__annotations__["return"], type)
>>> True
```

Reply:
Wait! further tests are problematic

Reply:
```
import typing

class A:
    pass

def func(x) -> typing.Type[A]:
    return A

isinstance(func.__annotations__["return"], type)
```

Reply:
The current implementation appears the safest.

Reply:
OK, @noc0lour, let's just stick with what you have. Could you still change the parenthesis back to the previous line?  You can add `[skip ci]` to the commit to avoid rerunning all the tests.  Happy to merge with that!